### PR TITLE
Py3 fixes

### PR DESCRIPTION
**Pull request by will-moore**
Enable Python 3 with unit tests passing on Python 2 and Python 3

Depends on:
 - https://github.com/ome/omero-py/pull/19
 - https://github.com/ome/omero-web/pull/10

Work in progress...

Run tests in docker with ```docker build -t web .; docker run -ti --rm web```.

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#73](https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/73/). See the [console output](https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/73/consoleText) for more details.
Possible conflicts:
  - PR #3 will-moore 'table download'
    - omeroweb/webclient/views.py
  - PR #10 will-moore 'unit tests'
    - omeroweb/webclient/views.py
  - PR #14 will-moore 'Django 1.10'
    - omeroweb/webclient/views.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#74](https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/74/). See the [console output](https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/74/consoleText) for more details.
Possible conflicts:
  - PR #3 will-moore 'table download'
    - omeroweb/webclient/views.py
  - PR #10 will-moore 'unit tests'
    - omeroweb/webclient/views.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#43](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/43/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/43/consoleText) for more details.
Possible conflicts:
  - PR #10 will-moore 'unit tests'
    - test/unit/test_util.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#18](https://idr-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/18/). See the [console output](https://idr-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/18/consoleText) for more details.
Possible conflicts:
  - PR #10 will-moore 'unit tests'
    - omeroweb/webgateway/urls.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#21](https://idr-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/21/). See the [console output](https://idr-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/21/consoleText) for more details.
Possible conflicts:
  - PR #10 will-moore 'unit tests'
    - omeroweb/webgateway/urls.py

--conflicts

File: omero/plugins/web.py [hunk 1]
```diff
@@ -304,7 +305,7 @@ def config(self, args, settings):
                          "wsgi or wsgi-tcp.")
 
         template_file = "%s.conf.template" % server
-        c = file(self._get_web_templates_dir() / template_file).read()
+        c = open(self._get_web_templates_dir() / template_file).read()
```
Comment (manics) [inline omero/plugins/web.py]: I had to change this line in https://github.com/ome/omero-web/pull/19/

**Pull request merged**